File: ai/entity/aplist/scenarioEntity.py
from sqlalchemy import Column, Integer, String, Text, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.ext.declarative import declarative_base

Base = declarative_base()

class ScenarioEntity(Base):
    __tablename__ = "scenario"

    key = Column(Integer, primary_key=True, autoincrement=True)  # 시나리오 고유 키
    id = Column(String(30), unique=True, nullable=False)  # 시나리오 아이디
    name = Column(String(50), nullable=False)  # 시나리오 이름
    description = Column(Text, nullable=False)  # 시나리오 설명
    validation = Column(Text, nullable=True)  # 검증 포인트
    flowChart = Column(Text, nullable=True)  # 흐름도
    projectKey = Column(Integer, ForeignKey("project.key"), nullable=False)  # 프로젝트 FK
    
    # 관계 설정
    project = relationship("Project", back_populates="scenarios")
    mappings = relationship("Mapping", back_populates="scenario", cascade="all, delete-orphan")
    
    def __init__(self, id: str, name: str, description: str, validation: str, flowChart: str, projectKey: int):
        self.id = id
        self.name = name
        self.description = description
        self.validation = validation
        self.flowChart = flowChart
        self.projectKey = projectKey
    

File: backend/spec/src/main/java/com/sk/skala/axcalibur/spec/feature/report/entity/ScenarioEntity.java
package com.sk.skala.axcalibur.spec.feature.report.entity;

import java.util.ArrayList;
import java.util.List;

import com.sk.skala.axcalibur.spec.global.entity.BaseTimeEntity;
import com.sk.skala.axcalibur.spec.global.entity.ProjectEntity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

/**
 * 시나리오 항목 테이블
 * 데이터베이스의 'scenario' 테이블과 매핑
 */
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "scenario")
public class ScenarioEntity extends BaseTimeEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "`key`")
    private Integer id;

    @Column(name = "id", nullable = false, unique = true)
    private String scenarioId;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(name = "validation", columnDefinition = "TEXT")
    private String validation;

    @Column(name = "flow_chart", columnDefinition = "TEXT")
    private String flowChart;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "project_key", nullable = false)
    private ProjectEntity projectKey;

    @OneToMany(mappedBy = "scenarioKey", cascade = CascadeType.REMOVE, orphanRemoval = true)
    @Builder.Default
    private List<MappingEntity> mappings = new ArrayList<>();
}